lib/Munin/Master/UpdateWorker.pm

```diff
@@ -421,70 +421,75 @@ sub uw_override_with_conf {
 }
 
 
+# Tune all RRD files if the munin version differs from the one which
+# wrote the data.
+#
+# Tune any RRD files where the metadata on disk differs from the
+# metadata received by munin-update.
 sub _compare_and_act_on_config_changes {
     my ($self, $nested_service_config) = @_;
 
-    # Kjellm: Why do we need to tune RRD files after upgrade?
-    # Shouldn't we create a upgrade script or something instead?
-    #
-    # janl: Upgrade script sucks.  This way it's inline in munin and
-    #  no need to remember anything or anything.
```

Comment:
I'd say that after a while in the trenches, I'm slowly in favor of upgrade scripts. Auto updating GAUGE/DERIVE seems a very bad idea after all. 

On the other side, updating the min/max looks nice. But I'm not sure that *ignoring* data is something that should look as innocent as it currently looks.